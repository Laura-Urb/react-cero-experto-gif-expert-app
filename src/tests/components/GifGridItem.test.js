import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
  const title = 'Titulo';
  const url =
    'https://media1.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=29e81697khpvl59ds9odzlf5el4cag89l64zqrksysp4dkyk&rid=giphy.gif&ct=g';

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de tener un pÃ¡rrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  test('debe de tener la imagen igual a la url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
  test('debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
