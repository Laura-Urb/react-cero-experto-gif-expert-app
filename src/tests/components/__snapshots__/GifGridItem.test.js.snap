// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://media1.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=29e81697khpvl59ds9odzlf5el4cag89l64zqrksysp4dkyk&rid=giphy.gif&ct=g"
  />
  <p>
    Titulo
  </p>
</div>
`;
